src/rust/engine/src/intrinsics.rs

```diff
@@ -213,16 +213,15 @@ fn remove_prefix_request_to_digest(
   let store = core.store();
 
   async move {
-    let input_digest = lift_digest(&externs::project_ignoring_type(&args[0], "digest"))?;
+    let input_digest =
+      lift_digest(&externs::project_ignoring_type(&args[0], "digest")).map_err(|e| throw(&e))?;
```

Comment:
Note that this change and most others are not necessary. The only thing we need is the fix for `download_file_to_digest` and `path_globs_to_digest`. 

But, personally, I think I find this refactor easier to follow and that it has less coercion of types, like no `let Result<_, String>`. Still, let me know if this is actually an improvement or I should revert to the old.

Reply:
This is only part of the refactor. The end effect should be the same, so it's only a question of what's clearer here.

I added comments on the two code blocks that are the actual fixes. Everything else can be reverted if preferred.

Pardon that I didn't make that clearer from the start, and thank you for the prompt feedback :)